### DISCUSSION TITLE
"Commit" vs "Force-close" vs "Non-collaborative settlement"

### DISCUSSION THREAD

**Discussion by thomaseizinger** [get10101/itchysats#741] [General]
We have a bit of a mixture of names around "Non-collaborative settlement". Here is a proposal for resolving this:

1. "Commit" should only refer to the commit transaction.
2. The action of publishing the commit transaction is _force closing_ a CFD.
3. "Non-collaborative settlement" should be removed from the codebase as it is the same as "force closing".

Input appreciated.

**Comment by klochowicz** [comment]
I'd like to pitch one more idea, that would also solve this issue: 
- "settle" instead of "close" (that would imply the fact that it's a collab settlement, and you actually need the other party, because you need to settle *with someone*)
- "close" to denote that you are actually closing the Cfd with the oracle attestation event at the time agreed when you opened.

then we could have "settling" and "closing" cfd, two short verbs, that could be rolled out it the codebase.

@thomaseizinger what do you think of it?

Going back to your original proposal, I agree that force-close is better than non-collaborative settlement. 👍

**Reply by thomaseizinger** [reply]
Thanks for the input!

"Closing" is nice because it is closer (pun intended) to the finance domain where you _close_ a position. That is why I would argue that "closing" should be the default with "force-closing" being the special case that is introduced by us having a blockchain underneath.

@bonomat Do you have an opinion here?

**Reply by klochowicz** [reply]
If this is closer to the to the domain jargon, then close and force-close work well for me!

**Reply by thomaseizinger** [reply]
I found this: https://www.investopedia.com/terms/c/closeposition.asp

So I think "closing" is a pretty good term here. What we are currently not modelling at all is the fact that closing a long position is performed by doing a sell.
Modelling this better could actually solve some problems here: https://github.com/itchysats/itchysats/discussions/1108 and https://github.com/itchysats/itchysats/discussions/829.

Closing a position requires placing an order that nullifies your current position. In order to place an order, you need a price. Currently, we are sending out offers from the maker. I think what we should be doing is sending out an offer for both going long AND going short. If the taker wants to close their long position, they would look at the maker's price feed for selling and place an order for that. How this order is getting filled is then a question of what we currently have open.

If the taker currently doesn't have an open position, selling is equivalent to opening a new short position.
If the taker has an open position with the exact same amount, selling will completely close this position.
If the taker has an open position with less than the order's amount, the long position will be turned into a short position for a margin of the difference.

@bonomat @da-kami Some thoughts on how we can (should?) implement going short.

**Reply by bonomat** [reply]
I think this is the right way:

> [...] I think what we should be doing is sending out an offer for both going long AND going short. If the taker wants to close their long position, they would look at the maker's price feed for selling and place an order for that. 

While this sounds like it opens up the door to partial closing, I would not tackle this for now, instead (for now) the user can only fully close his position (by taking a sell order). 

One thought which might be a challenge: 
Assuming we keep the same model for offers up with a `min` and `max` amount, this implies, that the taker always has a suitable sell order available. 
Differently said:  each maker who offers long also has to offer shorts. Otherwise their counterparty will not be able to close their position. 
What if this is not the case?

**Reply by klochowicz** [reply]
If the maker doesn't offer shorts, then the natural thing is to fall-back to force-close (that only means that closing ("collaborative settlement") would be impossible.

**Reply by bonomat** [reply]
Ok, makes sense. It could be in the interest of the maker to settle collaboratively though because it reduces on-chain transaction fees.

**Reply by klochowicz** [reply]
> Ok, makes sense. It could be in the interest of the maker to settle collaboratively though because it reduces on-chain transaction fees.

exactly, thus it would be in each maker's interest to allow short positions.

**Reply by bonomat** [reply]
I see that offering shorts has the positive side effect of letting people collab settle their longs but I'm not sure this is the right way. 
Imho we should try to separate these two things.

**Reply by thomaseizinger** [reply]
Here is another take:

Just because a maker sends out an offer, doesn't mean they have to accept every order that is coming in. A maker could easily accept orders that nullify positions with a taker (that is our current collab settlement protocol) but reject orders that would result in more short positions.

**Reply by bonomat** [reply]
I think this is a bit weird and adds an additional meaning into the offers. 
How would a taker distinguish offers from makers they are willing to take and offers they are only willing to take if it nullifies their position?

**Reply by thomaseizinger** [reply]
> I think this is a bit weird and adds an additional meaning into the offers.

Does it really? Or does it only appear this way because we have a special handling of closing at the moment? It seems to be the same on a domain level. Why should be special case this in the software?

>  How would a taker distinguish offers from makers they are willing to take and offers they are only willing to take if it nullifies their position?

You don't really and I also think there is no need to. We already have an accept/reject functionality so we have to deal with the case of an order being rejected. We also established recently that we actually want this accept/reject step. Why not leverage that?

**Comment by thomaseizinger** [comment]
Regarding the terminology of "force-close": Perhaps we should avoid that. https://www.investopedia.com/terms/c/closeposition.asp talks about a concept of "force-closing" under the heading "Special considerations".

**Reply by klochowicz** [reply]
I think you can still interpret our "force-closing" in similar way the link posted above understands it. The way I read it it means that they are closed by one side, without other's consent in certain situations- our lack of cooperation by the other party follows that understanding.

**Reply by thomaseizinger** [reply]
Good argument!
